pkgs/servers/fusiondirectory/default.nix

```diff
@@ -0,0 +1,129 @@
+{ lib, stdenv, fetchurl, lndir, schema2ldif, scriptaculous,
+prototypejs, smarty3, smarty3-i18n, gettext, makeWrapper,
+perl, perlPackages, openldap, plugins ? []}: stdenv.mkDerivation rec {
+  name = "fusiondirectory-${version}";
+  version = "1.2.1";
+
+  srcs = [
+    (fetchurl {
+      url = "https://repos.fusiondirectory.org/sources/fusiondirectory/fusiondirectory-${version}.tar.gz";
+      sha256 = "1pzrb3px0qn5rfwh29549lh7fgy46qcx2wlfzhcyax7anhx19ndx";
+    })
+    (fetchurl {
+      url = "https://repos.fusiondirectory.org/sources/fusiondirectory/fusiondirectory-plugins-${version}.tar.gz";
+      sha256 = "02n2wsp0dljw54wdd38yn0214v7kf218iaz0z9qw8ia1x0ffilnb";
+    })
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+  phases = [ "unpackPhase" "patchPhase" "installPhase" ];
+
+  unpackPhase = ''
+    mkdir plugins fusiondirectory
+    for src in $srcs; do
+      case "$src" in
+        *fusiondirectory-plugins*)
+          tar xf "$src" -C plugins --strip-components=1
+          ;;
+        *)
+          tar xf "$src" -C fusiondirectory --strip-components=1
+          ;;
+      esac
+    done
+  '';
+
+  prePatch = ''
+    # fusiondirectory-setup
+    sed -i \
+      -e 's:/usr/bin/perl:${perl}/bin/perl:g' \
+      -e "s:/var/www/fusiondirectory:$out:g" \
+      -e "s:/usr/share/php/smarty3:$out/share/smarty/Smarty.class.php:g" \
+      -e "s:/var/cache/fusiondirectory:$out/share/cache:g" \
+      -e 's:"msgcat:"${gettext}/bin/msgcat:g' \
+      -e 's:"msgfmt:"${gettext}/bin/msgfmt:g' \
+      -e '/use Term::ReadKey;/d' \
+      -e '/as root/d' \
+      fusiondirectory/contrib/bin/fusiondirectory-setup
+
+    # fusiondirectory-insert-schema
+    sed -i \
+      -e 's:/usr/bin/perl:${perl}/bin/perl:g' \
+      -e 's:ldap-schema-manager:${schema2ldif}/bin/ldap-schema-manager:g' \
+      -e "s:/etc/ldap/schema/fusiondirectory:$out/etc/schema:g" \
+      fusiondirectory/contrib/bin/fusiondirectory-insert-schema
+
+    # Path to ldapsearch
+    sed -i \
+      -e 's: = "ldapsearch: = "${openldap}/bin/ldapsearch:g' \
+      fusiondirectory/include/class_ldap.inc
+
+    # Path to fusiondirectory.auth
+    sed -i "s:CACHE_DIR:'/var/spool/fusiondirectory/':g" fusiondirectory/setup/class_setupStepWelcome.inc
+
+    # Plugin installation
+    sed 's/@@plugins@@/"${lib.concatStringsSep "\", \"" plugins}"/g' '${./setup-plugins.patch}' > setup-plugins.patch
+  '';
+
+  patches = [ "setup-plugins.patch" ];
```

Comment:
I think this should be `patches = [ ./setup-plugins.patch ]; `

Reply:
No, in the patch phase, I patch this patch and write it to $PWD (see line 64)

Reply:
To elaborate, I'd find it more intuitive if the patch was applied via the standard patch phase and the requisite substitutions performed in `postPatch`. You could reasonably argue that it works out to the same thing, but there is value in uniformity & sticking to the standard phases as much as possible. I at least am hesitant to merge stuff if I have to spend time trying to figure out how the recipe deviates from the standard boilerplate stuff, absent a clear rationale.